pkg/spoke/managedcluster/joining_controller.go

```diff
@@ -128,25 +128,32 @@ func (c *managedClusterJoiningController) getClusterResources() (capacity, alloc
 		return nil, nil, err
 	}
 
-	cpuCapacity := *resource.NewQuantity(int64(0), resource.DecimalSI)
-	memoryCapacity := *resource.NewQuantity(int64(0), resource.BinarySI)
-	cpuAllocatable := *resource.NewQuantity(int64(0), resource.DecimalSI)
-	memoryAllocatable := *resource.NewQuantity(int64(0), resource.BinarySI)
+	capacityList := make(map[clusterv1.ResourceName]resource.Quantity)
+	allocatableList := make(map[clusterv1.ResourceName]resource.Quantity)
+
 	for _, node := range nodes {
-		cpuCapacity.Add(*node.Status.Capacity.Cpu())
-		memoryCapacity.Add(*node.Status.Capacity.Memory())
-		cpuAllocatable.Add(*node.Status.Allocatable.Cpu())
-		memoryAllocatable.Add(*node.Status.Allocatable.Memory())
+		for key, value := range node.Status.Capacity {
+			if _, exist := capacityList[clusterv1.ResourceName(key)]; exist {
+				tmpQuantity := capacityList[clusterv1.ResourceName(key)]
+				tmpQuantity.Add(value)
+				capacityList[clusterv1.ResourceName(key)] = tmpQuantity
+			} else {
+				capacityList[clusterv1.ResourceName(key)] = value
```

Comment:
You may need copy the value to prevent from mutating the original one from node.